### PR TITLE
Ethereum PoC

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

**Comment by chfast** [comment]
## [Current coverage](https://codecov.io/github/imapp-pl/golem?ref=cba2bede958102cba4a2868850f8a3e19ccbb9db) is `36.53%`

> Merging **#93** into **develop** will decrease coverage by **-0.40%** as of [`cba2bed`](https://codecov.io/github/imapp-pl/golem/commit/cba2bede958102cba4a2868850f8a3e19ccbb9db)

``` diff
@@            develop     #93   diff @@
=======================================
  Files           252     254     +2
  Stmts         20745   20984   +239
  Branches       2328    2352    +24
  Methods           0       0       
=======================================
+ Hit            7662    7667     +5
  Partial         239     239       
- Missed        12844   13078   +234
```

> Review entire [Coverage Diff](https://codecov.io/github/imapp-pl/golem/compare/f326746016a39fa5cbd8c9e9777839539db9fb6e...cba2bede958102cba4a2868850f8a3e19ccbb9db) as of [`cba2bed`](https://codecov.io/github/imapp-pl/golem/commit/cba2bede958102cba4a2868850f8a3e19ccbb9db)

---

### [Uncovered Suggestions](https://codecov.io/github/imapp-pl/golem/features/suggestions?ref=cba2bede958102cba4a2868850f8a3e19ccbb9db)
1. `+0.12%` via [.../task/pbrtgnrtask.py#72...95](https://codecov.io/github/imapp-pl/golem/gnr/task/pbrtgnrtask.py?ref=cba2bede958102cba4a2868850f8a3e19ccbb9db#72) 
2. `+0.10%` via [golem/eth.py#1...20](https://codecov.io/github/imapp-pl/golem/golem/eth.py?ref=cba2bede958102cba4a2868850f8a3e19ccbb9db#1) 
3. `+0.09%` via [golem/eth.py#192...209](https://codecov.io/github/imapp-pl/golem/golem/eth.py?ref=cba2bede958102cba4a2868850f8a3e19ccbb9db#192) 
4. _[See 7 more...](https://codecov.io/github/imapp-pl/golem/features/suggestions?ref=cba2bede958102cba4a2868850f8a3e19ccbb9db)_

> Powered by [Codecov](https://codecov.io). Updated on successful CI builds.

**Comment by badb** [comment]
On windows there will be problems with levedb (required by pyethapp). There are few (but pretty old) unofficial forks that should build on windows, but they're not exactly working. I may try to search for some better implementations later.

**Comment by chfast** [comment]
I will handle Ethereum on Windows before merging that.

File: golem/eth.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    head = svc.chain.head
+    nonce = head.get_nonce(FAUCET_ADDR)
+    print "NONCE", nonce
+    if nonce == 0:  # Deploy Bank of Deposit contract
+        tx = Transaction(nonce, 1, 3141592, to='', value=0,
+                         data=BankOfDeposit.INIT_HEX.decode('hex'))
+        tx.sign(FAUCET_PRIVKEY)
+        svc.add_transaction(tx)
+        addr = tx.creates
+        assert addr == "cfdc7367e9ece2588afe4f530a9adaa69d5eaedb".decode('hex')
```
Comment (azawlocki) [inline golem/eth.py:271]: Why this address?

Comment (chfast) [inline golem/eth.py:271]: Contract's address is `keccak(sender, nonce)`.

**Comment by azawlocki** [comment]
Does it work on Windows? If so then please merge.

**Comment by chfast** [comment]
Not yet on Windows. I will need to work on that first.

**Comment by chfast** [comment]
Replaced by #144.

**Pull request closed without merging**